sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/addressEnumerator/AddressEnumerator.java

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.implementation.directconnectivity.addressEnumerator;
+
+import com.azure.cosmos.implementation.RxDocumentServiceRequest;
+import com.azure.cosmos.implementation.directconnectivity.Uri;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+import static com.azure.cosmos.implementation.directconnectivity.Uri.HealthStatus.Unhealthy;
+import static com.azure.cosmos.implementation.directconnectivity.Uri.HealthStatus.UnhealthyPending;
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+public class AddressEnumerator {
```

Comment:
Looks like there is not state in this class, why not create it in a helper class format with static methods?
It seems unnecessary to store state of this object throughout SDK. Thoughts?
Actually, similar to `AddressEnumeratorUsingPermutations`. Also, why do we need 2 separate classes? Is it such that in future we might use `AddressEnumeratorUsingSomethingElse` ? If so, then its okay, but if not, I would keep the AddressEnumeration logic at one place so it is easier to debug.

Reply:
yea, it makes sense to use static for AddressEnumerator

and yes, we have two: AddressEnumeratorFisherYateShuffle, and AddressEnumeratorUsingPermutations